src/postgres/commands/createPostgresServer.ts

```diff
@@ -0,0 +1,63 @@
+import { AzureWizard, AzureWizardPromptStep, IActionContext, ILocationWizardContext, LocationListStep, ResourceGroupListStep } from 'vscode-azureextensionui';
+import { ext } from '../../extensionVariables';
+import { SubscriptionTreeItem } from '../../tree/SubscriptionTreeItem';
+import { localize } from '../../utils/localize';
+import { nonNullProp } from '../../utils/nonNull';
+import { PostgresServerTreeItem } from '../tree/PostgresServerTreeItem';
+import { setFirewallRule } from './configurePostgresFirewall';
+import { IPostgresWizardContext } from './PostgresAccountWizard/IPostgresWizardContext';
+import { PostgresServerConfirmPWStep } from './PostgresAccountWizard/PostgresServerConfirmPWStep';
+import { PostgresServerCreateStep } from './PostgresAccountWizard/PostgresServerCreateStep';
+import { PostgresServerCredPWStep } from './PostgresAccountWizard/PostgresServerCredPWStep';
+import { PostgresServerCredUserStep } from './PostgresAccountWizard/PostgresServerCredUserStep';
+import { PostgresServerFirewallStep } from './PostgresAccountWizard/PostgresServerFirewallStep';
+import { PostgresServerNameStep } from './PostgresAccountWizard/PostgresServerNameStep';
+
+export async function createPostgresServer(context: IActionContext, node?: SubscriptionTreeItem): Promise<void> {
+    if (!node) {
+        node = await ext.tree.showTreeItemPicker<SubscriptionTreeItem>(SubscriptionTreeItem.contextValue, context);
+    }
+    const wizardContext: IPostgresWizardContext = Object.assign(context, node.root);
+    const promptSteps: AzureWizardPromptStep<ILocationWizardContext>[] = [
+        new PostgresServerNameStep(),
+        new ResourceGroupListStep(),
+        new PostgresServerCredUserStep(),
+        new PostgresServerCredPWStep(),
+        new PostgresServerConfirmPWStep(),
+        new PostgresServerFirewallStep()
+    ];
+
+    LocationListStep.addStep(wizardContext, promptSteps);
+    const wizard = new AzureWizard(wizardContext, {
+        promptSteps,
+        executeSteps: [
+            new PostgresServerCreateStep()
+        ],
+        title: localize('createPostgresServerPrompt', 'Create new PostgreSQL server')
+    });
+    await wizard.prompt();
+    const serverName: string = nonNullProp(wizardContext, 'newServerName');
+    await wizard.execute();
+
+    vscode.window.showInformationMessage(localize('createdServerMsg', 'Successfully created server "{0}".', wizardContext.newServerName));
+    const serverTree: PostgresServerTreeItem = new PostgresServerTreeItem(node, nonNullProp(wizardContext, 'server'));
+    let user: string = nonNullProp(wizardContext, 'adminUser');
+    const usernameSuffix: string = `@${serverName}`;
+    if (!user.includes(usernameSuffix)) {
+        user += usernameSuffix;
+    }
+    const password: string = nonNullProp(wizardContext, 'adminPassword');
+    void serverTree.setCredentials(user, password);
+    if (wizardContext.addFirewall) {
+        const ip: string = await publicIp.v4();
```

Comment:
Add `ip` to the wizard context and set it in `PostgresServerFirewallStep` so that you don't have to `await` this call again

Reply:
The setFirewallRule can only be called after we have the new Server Tree item created, so can't really set ip in PostgresServerFirewallStep. PostgresServer..Step is only prompting and setting the addFirewall boolean accordingly. But can add ip to setFirewallRule() method, I am not sure if that would help though

Reply:
I'm saying do something like this in `PostgresServerFirewallStep`:
```
wizardContext.publicIp = await publicIp.v4();
```

And whenever you call `setFirewallRule` (_not_ in `PostgresServerFirewallStep`), you would just use `wizardContext.publicIp` instead of calling `publicIp.v4` again

Reply:
ok so directly add it to wizardContext, makes sense... in this case we can remove it from configurePostgresFirewall as well